src/interfaces/simulation.ts

```diff
@@ -157,8 +135,8 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
             .catch(() => {
               throw new ZapperError("zap out approval transaction", ZapperError.ZAP_OUT_APPROVAL);
             })
-            .then(async (approvalTransaction) => {
-              return await this.simulateZapApprovalTransaction(approvalTransaction, options);
+            .then(async ({ from, to, data }) => {
+              return this.simulationExecutor.makeSimulationRequest(from, to, data, { ...options, save: true });
```

Comment:
why is `save: true` added? not sure what it does

Reply:
Sorry, that's very obscure yes. The reason is this https://github.com/yearn/yearn-sdk/blob/master/src/interfaces/simulation.ts#L507

I don't think `simulateZapApprovalTransaction()` was needed, so I've merged that into this bit of code and there was a `options.save = true;` which I've added to the options